docs/awsassumerolecreds.md

```diff
@@ -0,0 +1,107 @@
+# AWS AssumeRole credentials for clusters
+
+Currently ClusterDeployments for AWS require customers to provide Secret with static credentials.
+AWS best practices recommend against using static credentials in favour of temporary credential
+sources like IAM roles. The requirement for static credentials is enforced by OpenShift clusters,
+but in certain cases these requirements can be removed.
+
+OpenShift clusters with Manual AWS credential management like outlined in [AWS STS Provisioning](./aws-sts-provisioning.md)
+support using temporary credential for installing the clusters.
+
+To allow use AssumeRole credentials, the customer must trust AWS IAM entities like IAM roles in platform's
+AWS account to assume the role specified. The workflow looks like,
+
+`Some credentials source` -> assume role -> `platform service provider IAM role` -> assume role -> `customer IAM role`
+
+This allows customers to specify only the IAM role in ClusterDeployment and not provide static credentials.
+
+## Setting up IAM trust from customer to service provider
+
+Let's assume that the service provider IAM role is `arn:aws:iam::123456:role/hive-aws-service-provider`. The customer
+must allow this role to assume the role they want to use for ClusterDeployment.
+
+The assume role policy for the customer IAM role look like,
+
+```console
+$ cat customer-assume-role.json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": "arn:aws:iam::123456:role/hive-aws-service-provider"
+            },
+            "Action": "sts:AssumeRole"
+            "Condition": {"StringEquals": {"sts:ExternalId": "Unique ID Assigned by Platform"}}
+        }
+    ]
+}
+```
+
+Let's create an IAM role,
+
+```sh
+aws iam create-role --role-name customer-openshift-role --assume-role-policy-document file://customer-assume-role.json
+```
+
+Now add the required permissions to the IAM role `customer-openshift-role`,
+
+```sh
+aws iam put-role-policy --role-name customer-openshift-role --policy-name customer-openshift-role-policy --policy-document file://openshift-permissions.json
```

Comment:
the required permissions are whatever we has for the user that passed in credentialsSecretRef .. the IAM entity is different but the required permissions are the same.. So documenting them here is not appropriate.

Reply:
> --policy-document file://openshift-permissions.json

i was under the assumption that the invocation of the aws cli was good enough to explain that this is an IAM policy JSON document that contains permissions for OpenShift cluster management 🤷🏽 

updated to include a line to describe what this file is.

Reply:
The fact that it's an IAM policy JSON document is clear, the contents of it are not. (even now) It sounds like it would need to be the super set of everything the installer needs (release dependent, presumably from the OpenShift official docs), and everything Hive needs (undocumented but hopefully we get to that Jira soon).  :man_shrugging: